### ISSUE TITLE
The backbone is a ResNet50  and neck head yolov8  facing issue

### ISSUE THREAD

**Issue by SupriyaBorhade** [ultralytics/ultralytics#21491]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

nc: 80
depth_multiple: 0.33
width_multiple: 0.25

backbone:
  - [-1, 1, ResNet50Backbone, [True]]  # outputs: P3 (0), P4 (1), P5 (2)

head:
  - [2, 1, SPPF, [1024, 5]]                    # 3
  - [3, 1, nn.Upsample, [None, 2, "nearest"]]  # 4
  - [[4, 1], 1, Concat, [1]]                   # 5
  - [5, 3, C2f, [512]]                         # 6
  - [6, 1, nn.Upsample, [None, 2, "nearest"]]  # 7
  - [[7, 0], 1, Concat, [1]]                   # 8
  - [8, 3, C2f, [256]]                         # 9

  - [9, 1, Conv, [256, 3, 2]]                  # 10
  - [[10, 6], 1, Concat, [1]]                  # 11
  - [11, 3, C2f, [512]]                        # 12

  - [12, 1, Conv, [512, 3, 2]]                 # 13
  - [[13, 3], 1, Concat, [1]]                  # 14
  - [14, 3, C2f, [1024]]                       # 15

  - [[9, 12, 15], 1, Detect, [nc]]             # 16

(D:\conda_envs\yolov8_custom) D:\model2>python -m yolo_res.tests.test_full_model

✅ Testing Full Model Integration...

🔍 Backbone Output Shapes:
  Feature 0: torch.Size([1, 256, 160, 160])
  Feature 1: torch.Size([1, 512, 80, 80])
  Feature 2: torch.Size([1, 1024, 40, 40])

                   from  n    params  module                                       arguments
  0                  -1  1   8543296  yolo_res.models.backbone.ResNet50Backbone    [True]

❌ Model loading or forward pass failed: list index out of range

**Comment by UltralyticsAssistant** [comment]
👋 Hello @SupriyaBorhade, thank you for reaching out and sharing your custom backbone and head configuration with the Ultralytics team! 🚀 This is an automated response to help get you started—an Ultralytics engineer will assist you soon.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) so we can effectively debug the issue. This helps us better understand and replicate the error you’re experiencing.

If this is a custom training ❓ Question, please include as much context as possible, such as dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

You can also connect with the Ultralytics community for more support:
- For real-time chat, join [Discord](https://discord.com/invite/ultralytics) 🎧
- For deeper discussions, visit [Discourse](https://community.ultralytics.com/)
- Or join conversations on our [Subreddit](https://www.reddit.com/r/Ultralytics/)

## Upgrade

To ensure your issue isn’t already resolved, please upgrade to the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these up-to-date, verified environments (all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) are preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your patience and for helping improve Ultralytics YOLO! 💡

**Comment by Y-T-G** [comment]
The layer numbers are not correct. You need to start counting from 0. And fix the input layers based on the new number.
